### ISSUE TITLE
Node pool cost difference between Infracost and Azure calculator

### ISSUE THREAD

**Issue by alikhajeh1** [infracost/infracost#2416]
### Discussed in https://github.com/infracost/infracost/discussions/2336

<div type='discussions-op-text'>

<sup>Originally posted by **mloskot** March  7, 2023</sup>
I have a Terraform plan provisioning a new node pool for my AKS:

- VM size: Standard_E2as_v5 
- OS: Windows (set via `os_type` of [azurerm_kubernetes_cluster_node_pool](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster_node_pool))
- VM/Node count: 1
- Premium disk P10
- Pay as go
- Region: UK South

Here is the Infracost estimate calculated based on my Terraform plan:

```
+ module.aks_windows_node_pool.azurerm_kubernetes_cluster_node_pool.node_pools[3]
  +£84.93
    + Instance usage (pay as you go, Standard_E2as_v5)
      +£68.55
    + os_disk
        + Storage (P10, LRS)
          +£16.38
```

UPDATE: Suspecting `location` property of resources in my Terraform code may be not enough, I've added `INFRACOST_AZURE_OVERRIDE_REGION=uksouth` environment variable and Infracost gives a greater estimate - still lower than Azure calculator's:

```
+ module.aks_windows_node_pool.azurerm_kubernetes_cluster_node_pool.node_pools[4]
  +£101
    + Instance usage (pay as you go, Standard_E2as_v5)
      +£80.69
    + os_disk 
        + Storage (P10, LRS)
          +£19.82
```

If I go to https://azure.microsoft.com/en-gb/pricing/calculator/ and request estimate for the same components in the same region, then the total for VM + Disk is roughly twice higher, that the `£84.93` above:

![image](https://user-images.githubusercontent.com/80741/223424206-7a543cea-9064-410e-8da5-73c7b90f9800.png)

What is the reason of such a difference or where am I making the mistake?

</div>

**Comment by aliscott** [comment]
It looks like Infracost is always using the Linux pricing for AKS node pools.

**Comment by aliscott** [comment]
@mloskot this is now released in [v0.10.24](https://github.com/infracost/infracost/releases/tag/v0.10.24).
